modules/core/04-channel/keeper/grpc_query.go

```diff
@@ -327,6 +327,27 @@ func (q Keeper) PacketAcknowledgements(c context.Context, req *types.QueryPacket
 	acks := []*types.PacketState{}
 	store := prefix.NewStore(ctx.KVStore(q.storeKey), []byte(host.PacketAcknowledgementPrefixPath(req.PortId, req.ChannelId)))
 
+	// if a list of packet sequences is provided then query for each specific ack and return
+	// otherwise, maintain previous behaviour and perform paginated query
+	for _, seq := range req.PacketCommitmentSequences {
+		acknowledgementBz, found := q.GetPacketAcknowledgement(ctx, req.PortId, req.ChannelId, seq)
+		if !found || len(acknowledgementBz) == 0 {
+			return nil, status.Error(codes.NotFound, "packet acknowledgement hash not found")
+		}
```

Comment:
I think it is ok to just continue and don't append the ack to the list. Relayers want to use the return value to pass it into `UnreceivedAcknowledgements`. This will take a list of acks and check if they still need to be acknowledged on the source chain. So I think the relayers are expecting to pass in a list of commitments and get a len(acks) <= len(commitments) rather than an error

Reply:
Makes sense! Done